File: ios/Modules/Sources/VortexCrypto/api/KeyAgreement.swift
import Foundation

/// Asymmetric key agreement (X25519).
///
/// Responsibility ends at producing a raw 32-byte shared secret. HKDF or
/// any post-processing lives in [Kdf] — this split keeps the interface
/// swappable for Kyber / X448 without touching consumers.
public protocol KeyAgreement: Sendable {
    func generateKeyPair() -> KeyPair
    func agree(myPrivate: Data, theirPublic: Data) throws -> Data
}

public enum KeyAgreementError: Error {
    case invalidKeyLength(Int)
}

File: android/app/src/main/java/sol/vortexx/android/crypto/api/KeyAgreement.kt
package sol.vortexx.android.crypto.api

/**
 * Asymmetric key agreement (X25519 by default).
 *
 * Responsibility ends at producing a raw 32-byte shared secret from a
 * private / public key pair. HKDF or any other post-processing lives in
 * [Kdf] — that split keeps this interface swappable for Kyber / X448
 * without touching consumers.
 */
interface KeyAgreement {
    fun generateKeyPair(): KeyPair
    fun agree(myPrivate: ByteArray, theirPublic: ByteArray): ByteArray
}
